src/trie.py

```diff
@@ -0,0 +1,74 @@
+"""This module is an implementation of a trie tree.
+
+Words branch out from root, with root's immediate children being
+the initial letter of each word. Words can then branch from that initial,
+as well as from initial substrings.
+
+Methods include:
+contains(word): Check to see whether a word is in the tree. O(k), where k is the length of the given word.
+insert(word): Inserts a word into the trie tree. O(k), where k is the length of the given word.
+"""
+
+
+class TrieTree(object):
+    """A class for trie trees."""
+
+    def __init__(self):
+        """Instantiate an empty trie tree."""
+        self.root = {}
+        self._size = 0
+
+    def contains(self, word):
+        """Check whether a word is in the trie tree."""
+        this_node = self.root
+        for letter in word:
+            if letter in this_node:
+                this_node = this_node[letter]
+            else:
+                return False
+        if '$' in this_node:
+            return True
+        return False
+
+    def insert(self, word):
+        """Insert a word into the trie tree."""
+        this_node = self.root
+        if self.contains(word):
+            return
+        word += "$"
+        for letter in word:
+            if letter in this_node:
+                this_node = this_node[letter]
+            else:
+                this_node[letter] = {}
```

Comment:
Is there a built in method that you could leverage to either get the existing key, or add it if it does not exist?